04_numeronym/sirigithub/main.go

```diff
@@ -0,0 +1,26 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func numeronyms(vals ...string) []string {
+	result := make([]string, len(vals))
+	for i, val := range vals {
+		r := []rune(val)
+		runeLength := len(r)
+		if runeLength < 4 {
+			result[i] = val
+		} else {
+			result[i] = fmt.Sprintf("%c%d%c", r[0], runeLength-2, r[runeLength-1])
+		}
```

Comment:
just a suggestion, no need to make this change. I think this `if` block is more readable as:
```
if runeLength > 3 {
    val = fmt.Sprintf("%c%d%c", r[0], runeLength-2, r[runeLength-1])
}
result[i] = val
```

This captures that the only special thing being done is for strings with more than 3 runes and that it changes the value. As the assignment to the result slice is common, it is moved out of the conditional and done once.

Reply:
@camh-anz that is a neat trick thank you, i made the change.

Reply:
Done.